test/integration/features/cmd-addons.feature

```diff
@@ -0,0 +1,158 @@
+    Given Minishift has state "Does Not Exist"
+     When executing "minishift addons remove anyuid" succeeds
+     Then stdout should contain
+      """
+      No anyuid.addon.remove file found for anyuid add-on.
+      """
+
+  Scenario: Installing default add-ons manually
+     When executing "minishift addons install --defaults" succeeds
+     Then stdout should contain
+      """
+      Default add-ons anyuid, admin-user, xpaas, registry-route installed
+      """
+     When executing "minishift addons list" succeeds
+     Then stdout should contain "admin-user"
+     Then stdout should contain "anyuid"
+     Then stdout should contain "registry-route"
+     Then stdout should contain "xpaas"
+
+  @minishift-only
+  Scenario: Default addons are installed but not enabled by default
+     When executing "minishift addons list"
+     Then stdout should match "admin-user\s*: disabled\s*P\(0\)"
+      And stdout should match "anyuid\s*: disabled\s*P\(0\)"
+      And stdout should match "registry-route\s*: disabled\s*P\(0\)"
+      And stdout should match "xpaas\s*: disabled\s*P\(0\)"
+
+  Scenario: Enabling not installed add-on
+     When executing "minishift addons enable absent-addon"
+     Then stdout should contain
+      """
+      No add-on with the name absent-addon is installed.
+      """
+
+  Scenario: Enabling installed add-on
+     When executing "minishift addons enable xpaas" succeeds
+     Then stdout should contain "Add-on 'xpaas' enabled"
+      And executing "minishift addons list" succeeds
+      And stdout should match "xpaas\s*: enabled\s*P\(0\)"
+
+  Scenario: Enabling installed add-on with priority
+     When executing "minishift addons enable admin-user --priority 5" succeeds
+     Then stdout should contain "Add-on 'admin-user' enabled"
+      And executing "minishift addons list" succeeds
+      And stdout should match "admin-user\s*: enabled\s*P\(5\)"
+
+  Scenario: Starting Minishift with admin-user and xpaas add-ons enabled
+  The addons are being applied in correct order.
+    Given Minishift has state "Does Not Exist"
+     When executing "minishift start" succeeds
+     Then Minishift should have state "Running"
+      And stdout should match
+      """
+      Applying addon 'xpaas'[\S\s]*Applying addon 'admin-user'
+      """
+      And stdout should not contain "Applying addon 'registry-route'"
+      And stdout should not contain "Applying addon 'any-uid'"
+
+  Scenario: Disabling enabled add-on
+     When executing "minishift addons disable xpaas" succeeds
+     Then stdout should contain
+      """
+      Add-on 'xpaas' disabled
+      """
+     When executing "minishift addons list" succeeds
+     Then stdout should match "xpaas\s*: disabled"
+
+  Scenario: Disabling disabled add-on
+     When executing "minishift addons disable xpaas" succeeds
+     Then stdout should contain
+      """
+      Add-on 'xpaas' disabled
+      """
```

Comment:
Nitpick : I feel the output from minishift should be different e.g. `Add-on 'xpaas' is already in disabled state`

Reply:
@LalatenduMohanty +1 That would be nicer, seems like easy-fix, will create an issue for that.

Reply:
It would be an additional check that does not really solve a problem.

  * If set, we do not take action
  * we just do as requested

Either way, we do not break the user interface design rule of "providing the user feedback". We perform the expected action. Indicating it was already in that state, does not change the outcome.